proposals/imageverificationpolicy.md

```diff
@@ -0,0 +1,289 @@
+
+```yaml
+images:
+- name: containers
+ expression: request.object.spec.containers.image
+- name: initContainers
+ expression: request.object.spec.initContainers.image
+- name: ephemeralContainers
+ expression: request.object.spec.ephemeralContainers.image
+- name: all
+ expression: request.object.spec.{containers,initContainers.ephemeralContainers}.image
+```
+
+## Attestors
+Attestors are the trusted identities, keys and certificates, i.e. trusted authorities. They can be of several types, like Cosign Key, Keyless, KMS or Notary's certs. It can also be extended in the future for newer attestors. Attestors can be accessed in verification block using `attestors.<name>`. Attestors and Attestations serve different purposes. Attestors are trust information Attestations are additional image metadata that should be verified, separating them simplifies the structure. The relationship between attestors and attestations is described in `verification` block.
+
+Attestors and Attestations are separated from each other, unlike current schema where users can specify `verifyImages.attestors` to verify only images and `verifyImages.attestations.attestors` to verify images and attestations which is complex and too nested. That can lead to confusion and makes policies hard to read because of nested YAML syntax.
+
+```yaml
+attestors:
+- name: keyless
+  cosign:
+    keyless:
+     identities:
+     - issuerRegExp: .*kubernetes.default.*
+       subjectRegExp: .*kubernetes.io/namespaces/default/serviceaccounts/default
+    ctlog:
+      url: http://rekor.rekor-system.svc
+      ignoreTlog: false
+- name: kms
+  cosign:
+    key:
+      kms: awskms:///arn:aws:kms:us-east-1:123456789012:alias/cosign-key
+```
+
+## Attestations
+Attestations are additional image metadata attached to an image by different methods such as In-toto Attestations or OCI 1.1 Referrers API. these attestations are fetched from the registry and then can be verified against the attestors. Their payload can also be accessed in verification block as json objects using `attestations.<name>` to run CEL expressions against them.
+```yaml
+attestations:
+- name: slsa
+  intoto:
+    type: https://slsa.dev/provenance/v0.2
+- name: sbom
+  referrer:
+    type: sbom/cyclone-dx
+```
+
+## Verifications
+Verifications are similar to validation policies' `spec.validation`. They can be used to specify CEL expressions to perform image verification, or attestation payload validation. The entire admission request is also present in the context as `request` object and CEL expressions can be written to verify that.
+
+```yaml
+verifications:
+- expression: >-
+    verifyImageSignatures(images.initContainers[0], attestors.kms)
```

Comment:
Typically users will want to verify all images in a pod are signed with an attestor and may want to exclude a certain references e.g. `docker.io/istio*`.

How would they do that?